### ISSUE TITLE
Created file type not getting shared in the channel 

### ISSUE THREAD

**Issue by AayushChaudhary0001** [mattermost-community/mattermost-plugin-google-drive#1]
During testing of this plugin, I found a bug which is not consistent.

When a user runs the command to create a doc/slide/sheet, and checks the checkbox to share it to the channel as well, the file type is not always shared on the channel.

**Steps to reproduce:-**
- Run the "/drive create < desired file type >" .
- Fill the mandatory fields and check the channel sharing checkbox.
- Open the channel where the file was shared.

The shared file is not getting reflected in the channel

Note:- This bug is not consistent, you might not be able to reproduce it every time.

**Comment by darkLord19** [comment]
Can you share some logs or approx steps which can help me reproduce this issue? @AayushChaudhary0001

**Comment by wiggin77** [comment]
@AayushChaudhary0001 can you provide the information requested?

**Comment by AayushChaudhary0001** [comment]
@darkLord19 I tried to reproduce this and I have created a video for the same, which I am attaching below, please have a look.


https://github.com/user-attachments/assets/845d125f-8028-4c79-a245-f2037cc48586


I am also adding the logs which I am receiving below.

{"timestamp":"2024-07-31 11:08:44.975 Z","level":"error","msg":"something went wrong while updating permissions for file","caller":"app/plugin_api.go:940","plugin_id":"com.darklord.plugin-google-drive","err":"googleapi: Error 403: The user does not have sufficient permissions for this file., insufficientFilePermissions","fileID":"1pLyqJJCmAfoPWNBRo05ayLkRtDGlD6YiM-nbljbFuEw"}
{"timestamp":"2024-07-31 11:08:45.259 Z","level":"error","msg":"Owner of the file performed this action","caller":"app/plugin_api.go:940","plugin_id":"com.darklord.plugin-google-drive","pageToken":"94617"}
{"timestamp":"2024-07-31 11:08:44.976 Z","level":"error","msg":"failed to modify file permissions","caller":"app/plugin_api.go:940","plugin_id":"com.darklord.plugin-google-drive","err":"googleapi: Error 403: The user does not have sufficient permissions for this file., insufficientFilePermissions"}


Please have a look at these of they can help, also let me know if I am missing something from my end.

**Comment by darkLord19** [comment]
@AayushChaudhary0001 can you share which apis have you enabled for you gcp project? Also can you tell me if you are able to update permission of file to `anyone` manually for the doc using the same account you have used for mm connection?

**Comment by AayushChaudhary0001** [comment]
@darkLord19 Yes I am able to update the permission type manually for the doc using the same account.

**Comment by darkLord19** [comment]
@AayushChaudhary0001 can you pull this branch https://github.com/mattermost-community/mattermost-plugin-google-drive/pull/11 and use it to test this feature again? Also clear all the data from previous plugin, disconnect and connect your account again

**Comment by BenCookie95** [comment]
@AayushChaudhary0001 Can you confirm what apis were enabled when this was failing? This is difficult to reproduce. 

Did the branch from @darkLord19 resolve the issue for you?

**Comment by AayushChaudhary0001** [comment]
@darkLord19 @BenCookie95 This branch #11 seems to be deleted without merge, can you please let me know form where I can re-test this issue?
